### PR TITLE
Add rust test creating and building a project

### PR DESCRIPTION
**Pull request by kalikiana**
A new project is created and built.

- Related ticket: https://progress.opensuse.org/issues/101073
- Verification run: https://openqa.opensuse.org/tests/2064752#step/rust/9

### CODE REVIEW

File: tests/console/rust.pm [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# SUSE's openQA tests
+#
+# Copyright 2019 SUSE LLC
```
Comment (okurz) [inline tests/console/rust.pm:3]: Please just remove the year

```suggestion
# Copyright SUSE LLC
```

File: tests/console/rust.pm [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    select_console('root-console');
+
+    zypper_call('in cargo');
+    assert_script_run('cargo new testproject');
+    assert_script_run(qq(echo 'uuid = "0.8"' >> testproject/Cargo.toml));
+    assert_script_run('cargo run --manifest-path testproject/Cargo.toml');
+}
+
+sub test_flags {
+    return {fatal => 1};
```
Comment (okurz) [inline tests/console/rust.pm:28]: No need to make this fatal, remove test_flags please

File: tests/console/rust.pm [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+use warnings;
+use testapi;
+use repo_tools;
+use version_utils;
+
+sub run {
+    select_console('root-console');
+
+    zypper_call('in cargo');
+    assert_script_run('cargo new testproject');
```
Comment (okurz) [inline tests/console/rust.pm:22]: Please try that as non-root, so add `select_console 'user-console';`

File: tests/console/rust.pm [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+# Summary: Create and run a project
+#    Use cargo to create a project which pulls in dependencies online.
+# Maintainer: Cris Dywan <cdywan@suse.com>
+
+use base "consoletest";
+use strict;
+use warnings;
+use testapi;
+use repo_tools;
+use version_utils;
```
Comment (okurz) [inline tests/console/rust.pm:16]: Looks unused, please remove

**Review by okurz — CHANGES_REQUESTED** [review]

**Comment by okurz** [comment]
You referenced a dummy verification run. Do you have a real one by now?

**Review by okurz — COMMENTED** [review]
ok, VR looks good. But this is a 15.3 test and according to schedule/functional/extra_tests_console you are adding it for Tumbleweed as well I assume. Please provide a VR for that as well

File: tests/console/rust.pm (hunk 1 above)
Comment (b10n1k) [reply tests/console/rust.pm]: Can i ask why? We use the date everywhere so i am not sure why this is an exception

Comment (okurz) [reply tests/console/rust.pm]: That's not an exception but the rule for *new* files, see https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/CONTRIBUTING.md#coding-style . Only for existing files it should be kept as is. I assume other reviewers do not look out for that in all cases.

File: lib/main_common.pm [hunk 5]
```diff
@@ -1055,6 +1055,7 @@ sub load_console_server_tests {
         loadtest "console/yast2_nfs_server";
     }
     loadtest "console/rsync";
+    loadtest "console/rust";
```
Comment (okurz) [inline lib/main_common.pm:1058]: IMHO wrong location. Should be in "extra tests" only. https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/13756/files#diff-051cb1f26a96a4f724250d9182535d868c691e20963ef91a1a7ca109e9538379R1609 and below

**Comment by kalikiana** [comment]
> ok, VR looks good. But this is a 15.3 test and according to schedule/functional/extra_tests_console you are adding it for Tumbleweed as well I assume. Please provide a VR for that as well

    ./script/openqa-clone-job --within-instance https://openqa.opensuse.org --skip-deps 2063495 _GROUP=0 CASEDIR="https://github.com/kalikiana/os-autoinst-distri-opensuse.git#rust.pm" SCHEDULE=tests/boot/boot_to_desktop,tests/console/rust
    https://openqa.opensuse.org/t2064645

I assume this is correct. Unfortunately none of my jobs is getting picked up now :-D

**Comment by okurz** [comment]
> > ok, VR looks good. But this is a 15.3 test and according to schedule/functional/extra_tests_console you are adding it for Tumbleweed as well I assume. Please provide a VR for that as well
> 
> ```
> ./script/openqa-clone-job --within-instance https://openqa.opensuse.org --skip-deps 2063495 _GROUP=0 CASEDIR="https://github.com/kalikiana/os-autoinst-distri-opensuse.git#rust.pm" SCHEDULE=tests/boot/boot_to_desktop,tests/console/rust
> https://openqa.opensuse.org/t2064512
> ```

Better would have been to also overwrite `TEST`. Because now your "custom" test shows up along with the production ones in https://openqa.opensuse.org/tests/2064512#next_previous

We have openqa-clone-custom-git-refspec for that use case. The equivalent to the above with a proper test name automatically provided would be as simple as

```
openqa-clone-custom-git-refspec https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/13756 https://openqa.opensuse.org/tests/2063495 SCHEDULE=tests/boot/boot_to_desktop,tests/console/rust
```

> I assume this is correct. Unfortunately none of my jobs is getting picked up now :-D

Finished by now. https://openqa.opensuse.org/tests/2064512#step/boot_to_desktop/2 fails because that is a scenario that still has an ISO but your schedule changes make the test not expect that. And there is no qcow image provided so there is nothing to boot from then. As previously shown a better base for cloning would be one of the existing extra tests, e.g. from the list https://openqa.opensuse.org/tests/overview?distri=opensuse&version=Tumbleweed&groupid=1&arch=x86_64 pick any "extra". With that the above can become

```
openqa-clone-custom-git-refspec https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/13756 https://openqa.opensuse.org/tests/2063701 SCHEDULE=tests/boot/boot_to_desktop,tests/console/rust
```

-> https://openqa.opensuse.org/tests/2064752

passed, there you go :)

**Review by okurz — APPROVED** [review]

**Pull request merged**